File: old/Taxi/1.rb
limit = gets.to_i
x = gets.split(" ").map{|x| x.to_i}
x =x.sort.reverse
i=0
j=limit-1
ans =0
while i<=j
	ans+=1
	four = 4-x[i]
	while x[j]<=four and j>=i
		four-=x[j]
		j-=1
	end
	i+=1
end
puts ans

File: old/Taxi/1.py
limit = int(raw_input())
x = raw_input("").split(" ")
x.sort(reverse=True)
x = [int(i)for i in x ]
j = limit-1
ans = 0
i = 0
while i<=j:
	ans+=1
	four = 4-x[i]
	while x[j]<=four and j>=i:
		four-=x[j]
		j-=1
	i+=1
print ans